chore: replace ↦ with =>

semorrison opened the pull request:
Additional proposal:

* We still use `↦` in documentation. Let's change to some different symbol, so one can simply search for `↦` and know that every appearance is "wrong".
* Lint for this.

digama0 commented:
I'm unclear on your first point. Did you replace all occurrences of `↦` in docs or not? I see one replacement in docs here but I'm not sure if this is all of them.

semorrison commented:
There are lots of docs regarding HashMap that use this symbol, e.g.

```
/--
Similar to `insert`, but also returns a boolean flag indicating whether an existing entry has been
replaced with `a ↦ b`.
-/
```

The one use in documentation that I replaced here was the "standard" (Mathlib) meaning.

digama0 commented:
Hm, I see. I suppose that it wouldn't be that bad to just use `=>` in those comments as well.

semorrison commented:
Okay, in that case I will update this PR to use `=>` everywhere, and then think about adding a linter.
